wallet/src/de/schildbach/wallet/BaseWalletApplication.kt

```diff
@@ -60,29 +60,19 @@ abstract class BaseWalletApplication : MultiDexApplication(), WalletDataProvider
         return getWalletData()!!.freshReceiveAddress()
     }
 
-    override fun getExchangeRate(currencyCode: String): LiveData<ExchangeRate> {
+    override fun getExchangeRate(currencyCode: String): LiveData<ExchangeRateData> {
         return ExchangeRatesRepository.instance.getRate(currencyCode).switchMap {
-            return@switchMap MutableLiveData<ExchangeRate>().apply {
-                value = ExchangeRate(it.currencyCode, it.rate, it.getCurrencyName(this@BaseWalletApplication), it.fiat)
+            return@switchMap MutableLiveData<ExchangeRateData>().apply {
+                value = ExchangeRateData(it.currencyCode, it.rate ?: "0.0", it.getCurrencyName(this@BaseWalletApplication), it.fiat)
             }
         }
     }
 
-    override fun getExchangeRates(): LiveData<List<ExchangeRate>> {
+    override fun getExchangeRates(): LiveData<List<ExchangeRateData>> {
         return ExchangeRatesRepository.instance.rates.switchMap {
-            return@switchMap MutableLiveData<List<ExchangeRate>>().apply {
+            return@switchMap MutableLiveData<List<ExchangeRateData>>().apply {
                 value = it.map { exchangeRate ->
-                    ExchangeRate(exchangeRate.currencyCode, exchangeRate.rate, exchangeRate.getCurrencyName(this@BaseWalletApplication), exchangeRate.fiat)
-                }
-            }
-        }
-    }
-
-    override fun currencyCodes(): LiveData<List<String>> {
```

Comment:
`currencyCodes()` method wasn't used. If needed, it's better to re-add as `Flow<List<String>>`